### ISSUE TITLE
An illegal memory access was encountered

### ISSUE THREAD

**Issue by PkuRainBow** [facebookresearch/maskrcnn-benchmark#45]
## To Reproduce

1.
1.
1.

## Environment

Please copy and paste the output from the
[environment collection script from PyTorch](https://raw.githubusercontent.com/pytorch/pytorch/master/torch/utils/collect_env.py)
(or fill out the checklist below manually).

You can get the script and run it with:
```
wget https://raw.githubusercontent.com/pytorch/pytorch/master/torch/utils/collect_env.py
# For security purposes, please check the contents of collect_env.py before running it.
python collect_env.py
```

 - PyTorch Version (e.g., 1.0):
 - OS (e.g., Linux):
 - How you installed PyTorch (`conda`, `pip`, source):
 - Build command you used (if compiling from source):
 - Python version:
 - CUDA/cuDNN version:
 - GPU models and configuration:
 - Any other relevant information:

**Comment by fmassa** [comment]
Could you give more information?

I suspect it happens because you used a too high learning rate, and training diverged, giving large indices.

**Comment by fmassa** [comment]
So, you have changed the `IMS_PER_BATCH` to be 2, and the learning rate as well?

**Comment by fmassa** [comment]
Try following the learning rate adaptation rules that I mentioned in the README, they are necessary for training to not diverge

**Comment by PkuRainBow** [comment]
@fmassa I still can not figure the problem.

**Comment by fmassa** [comment]
So, to double check:
- you are using 4 GPUs
- you set `IMS_PER_BATCH` to 2

Is that right?

Note that the meaning of `IMS_PER_BATCH` is different in `maskrcnn-benchmark` than it is from `Detectron`.
If you use fewer GPUs than 8, then you might need to change s few hyper parameters for training to behave the same.
Have a look at https://github.com/facebookresearch/maskrcnn-benchmark#single-gpu-training for the differences and what to do.

Let me know if you still have problems

**Comment by PkuRainBow** [comment]
@fmassa Thanks for your kind help.

I will update if I have got progress.

**Comment by zimenglan-sysu-512** [comment]
hi @fmassa 
after several thousands iterations or several tens of thousands iterations, the loss become `NaN`, do u have ideas to solve it?

**Comment by fmassa** [comment]
@zimenglan-sysu-512 difficult to say without more context. Is this COCO? Are you using a standard model or have you adapted one of the models? It might require some digging to understand where the problem might come from.

**Comment by zimenglan-sysu-512** [comment]
hi @fmassa 
i want to add `light-head rcnn` to train `R-50-C4` on COCO dataset, maybe something wrong in my code to implement it. i need to check my code.
thanks.
